src/test/java/net/imagej/convert/NumberAndTypeConversionTest.java

```diff
@@ -50,77 +58,138 @@
+				{ 0, new UnsignedShortType((byte) 0) }, //
+				{ BigInteger.ZERO, new UnsignedLongType((byte) 0) }, //
+				{ (byte) 0, new ByteType((byte) 0) }, //
+				{ 0, new IntType(0) }, //
+				{ 0L, new LongType(0L) }, //
+				{ (short) 0, new ShortType((short) 0) }, //
+				{ 0., new DoubleType(0.) }, //
+				{ 0.F, new FloatType(0f) }, //
+			});
+		}
 
-	@Parameter(1)
-	public Object numericType;
+		@Parameter(0)
+		public Object number;
 
-	private ConvertService convertService;
+		@Parameter(1)
+		public Object numericType;
 
-	@Before
-	public void setUp() {
-		final Context ctx = new Context();
-		convertService = ctx.getService(ConvertService.class);
-	}
+		private ConvertService convertService;
 
-	@After
-	public void tearDown() {
-		convertService.getContext().dispose();
-		convertService = null;
-	}
+		@Before
+		public void setUp() {
+			final Context ctx = new Context();
+			convertService = ctx.getService(ConvertService.class);
+		}
+
+		@After
+		public void tearDown() {
+			convertService.getContext().dispose();
+			convertService = null;
+		}
+
+		/**
+		 * Test conversion from {@link Number} to {@link NumericType}
+		 */
+		@SuppressWarnings({ "rawtypes", "unchecked" })
+		@Test
+		public void testNumberToNumericType() {
+			assertTrue("Conversion support from number to numeric type",
+				convertService.supports(number, numericType.getClass()));
+			ValueEquals converted = (ValueEquals) convertService.convert(number,
+				numericType.getClass());
+			assertTrue("Converted type value equality", converted.valueEquals(
+				numericType));
+		}
 
-	/**
-	 * Test conversion from {@link Number} to {@link NumericType}
-	 */
-	@SuppressWarnings({ "rawtypes", "unchecked" })
-	@Test
-	public void testNumberToNumericType() {
-		assertTrue("Conversion support from number to numeric type", convertService
-			.supports(number, numericType.getClass()));
-		ValueEquals converted = (ValueEquals) convertService.convert(number,
-			numericType.getClass());
-		assertTrue("Converted type value equality", converted.valueEquals(
-			numericType));
+		/**
+		 * Test conversion from {@link Number} to {@link NumericType}
+		 */
+		@Test
+		public void testNumericTypeToNumber() {
+			assertTrue("Conversion support from numeric type to number",
+				convertService.supports(numericType, number.getClass()));
+			Number converted = (Number) convertService.convert(numericType, number
+				.getClass());
+			assertEquals("Converted value equality", number, converted);
+		}
 	}
 
-	/**
-	 * Test conversion from {@link Number} to {@link NumericType}
-	 */
-	@Test
-	public void testNumericTypeToNumber() {
-		assertTrue("Conversion support from numeric type to number", convertService
-			.supports(numericType, number.getClass()));
-		Number converted = (Number) convertService.convert(numericType, number
-			.getClass());
-		assertEquals("Converted value equality", number, converted);
+	public static class NumberToRealTypeTest {
```

Comment:
This could have been a separate `.java` file, and thus avoided the need to use the experimental `Enclosed`, couldn't it? Or is there shared state between the two inner classes?